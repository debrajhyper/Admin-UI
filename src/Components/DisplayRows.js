import React from 'react'

import { FaUserEdit, FaTrash } from 'react-icons/fa'

const DisplayRows = ({ user, users, setUsers, setEditUserId, setEditUserForm }) => {

    //  Check if the user is checked and set the users to the state using the setUsers() method.
    const handleChecked = e => {
        const { id, value, checked } = e.target;
        let newUsers = users.map(user => user.id === id && user.name === value ? { ...user, isChecked: checked } : user);
        setUsers(newUsers);
    };

    //  Edit the user data and set the user data to the state using the setEditUserForm() method.
    const handleUserEdit = (e, user) => {
        e.preventDefault();
        setEditUserId(user.id);
        const formValues = {
            id: user.id,
            name: user.name,
            email: user.email,
            role: user.role
        };
        setEditUserForm(formValues);
    };
    
    //  Delete the user from the users array and set the users to the state using the setUsers() method.
    const handleUserDelete = userId => {
        const newUsers = [ ...users ];
        const index = users.findIndex(user => user.id === userId);
        newUsers.splice(index, 1);
        setUsers(newUsers);
    };

    return (
        <tr id={user.id} className={`data-row bg-white whitespace-nowrap hover:cursor-default hover:bg-blue-light hover:text-white ${user?.isChecked ? 'bg-gray-100' : null}`}>
            <td className="px-6 py-2 whitespace-nowrap text-center">
                <input 
                    type="checkbox" 
                    className="form-checkbox h-4 w-4 outline-none focus:outline-none" 
                    id={user.id}
                    onChange={handleChecked}
                    checked={user?.isChecked || false}
                    value={user.name}
                />
            </td>
            <td className="px-6 py-2 whitespace-nowrap">{user.name}</td>
            <td className="px-6 py-2 whitespace-nowrap">{user.email}</td>
            <td className="px-6 py-2 whitespace-nowrap">{user.role}</td>
            <td className="px-6 py-2 whitespace-nowrap text-center">
                <button 
                    type="button" 
                    title="Edit" 
                    className="btn p-2 mr-2 text-blue-light rounded" 
                    onClick={(e) => handleUserEdit(e, user)} 
                >
                    <FaUserEdit size={16} />
                </button>
                <button 
                    type="button" 
                    title="Delete" 
                    className="btn p-2 ml-2 text-red-500 rounded" 
                    onClick={() => handleUserDelete(user.id)} 
                >
                    <FaTrash size={16} />
                </button>
            </td>
        </tr>
    )
}

export default DisplayRows
