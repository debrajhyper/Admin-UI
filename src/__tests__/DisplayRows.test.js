import { render, screen, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import DisplayRows from "../Components/DisplayRows";

afterEach(cleanup);

test("render without crashing", () => {
    const users = [
        { id: 1, name: "Aaron Miles", email: "aaron@mailinator.com", role: "member" },
        { id: 2, name: "Aishwarya Naik", email: "aishwarya@mailinator.com", role: "member" },
        { id: 3, name: "Arvind Kumar", email: "arvind@mailinator.com", role: "admin" },
    ];
    render(<DisplayRows user={users[0]} users={users} />);

    const TrElement = screen.getByTestId("display-rows");
    expect(TrElement).toBeInTheDocument();
    expect(TrElement).toContainHTML("<tr>");
    expect(TrElement).toContainHTML("<td>");
})

test("snapshot", () => {
    const users = [
        { id: 1, name: "Aaron Miles", email: "aaron@mailinator.com", role: "member" },
        { id: 2, name: "Aishwarya Naik", email: "aishwarya@mailinator.com", role: "member" },
        { id: 3, name: "Arvind Kumar", email: "arvind@mailinator.com", role: "admin" },
    ];

    const tree = renderer.create(<DisplayRows user={users[0]} users={users} />).toJSON();
    expect(tree).toMatchSnapshot();
})