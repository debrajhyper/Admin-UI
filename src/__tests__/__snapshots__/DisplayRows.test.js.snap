// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<tr
  className="data-row bg-white dark:bg-dark whitespace-nowrap hover:cursor-default hover:bg-blue-light dark:hover:bg-blue-dark hover:text-white null"
  data-testid="display-rows"
  id={1}
>
  <td
    className="px-6 py-2 whitespace-nowrap text-center"
  >
    <input
      checked={false}
      className="form-checkbox h-4 w-4 outline-none focus:outline-none"
      id={1}
      onChange={[Function]}
      type="checkbox"
      value="Aaron Miles"
    />
  </td>
  <td
    className="px-6 py-2 whitespace-nowrap text-left"
  >
    Aaron Miles
  </td>
  <td
    className="px-6 py-2 whitespace-nowrap text-left"
  >
    aaron@mailinator.com
  </td>
  <td
    className="px-6 py-2 whitespace-nowrap text-left"
  >
    member
  </td>
  <td
    className="px-6 py-2 whitespace-nowrap text-center"
  >
    <button
      className="btn p-2 mr-2 text-blue-light rounded dark:text-blue-dark"
      onClick={[Function]}
      title="Edit"
      type="button"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 640 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h274.9c-2.4-6.8-3.4-14-2.6-21.3l6.8-60.9 1.2-11.1 7.9-7.9 77.3-77.3c-24.5-27.7-60-45.5-99.9-45.5zm45.3 145.3l-6.8 61c-1.1 10.2 7.5 18.8 17.6 17.6l60.9-6.8 137.9-137.9-71.7-71.7-137.9 137.8zM633 268.9L595.1 231c-9.3-9.3-24.5-9.3-33.8 0l-37.8 37.8-4.1 4.1 71.8 71.7 41.8-41.8c9.3-9.4 9.3-24.5 0-33.9z"
        />
      </svg>
    </button>
    <button
      className="btn p-2 ml-2 text-red-500 rounded"
      onClick={[Function]}
      title="Delete"
      type="button"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 448 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
        />
      </svg>
    </button>
  </td>
</tr>
`;
